packages/core/src/types.ts

```diff
@@ -472,26 +472,31 @@ export type TransitionsConfig<
   TAction extends ParameterizedObject,
   TGuard extends ParameterizedObject,
   TDelay extends string
-> = {
-  // TODO: this doesn't support partial descriptors
-  [K in TEvent['type'] | '*']?: K extends '*'
-    ? TransitionConfigOrTarget<
-        TContext,
-        TEvent,
-        TEvent,
-        TAction,
-        TGuard,
-        TDelay
-      >
-    : TransitionConfigOrTarget<
+> =
+  | {
+      [K in
+        | TEvent['type']
+        | PartialEventType<TEvent['type']>
+        | '*']?: TransitionConfigOrTarget<
         TContext,
         ExtractEvent<TEvent, K>,
         TEvent,
         TAction,
         TGuard,
         TDelay
       >;
-};
+    };
+
+type PartialEventType<TEventType extends string> =
+  TEventType extends `${infer TEventPrefix}.${infer TEventSuffix}`
+    ? `${TEventPrefix}.*` | `${TEventPrefix}.${PartialEventType<TEventSuffix>}`
+    : never;
```

Comment:
A version of this that would leverage tail-recursion elimination could look smth like this:
```suggestion
type _PartialEventType<
  TEventType extends string,
  Consumed extends string = "",
  Acc extends string = never,
> = TEventType extends `${Consumed}${infer TEventPrefix}.${string}`
  ? _PartialEventType<
      TEventType,
      `${Consumed}${TEventPrefix}.`,
      Acc | `${Consumed}${TEventPrefix}.*`
    >
  : Acc;

type PartialEventType<TEventType extends string> = _PartialEventType<TEventType>
```

It's hard to imagine somebody creating an event type with more than 50 segments though... and this definitely looks more complex 😅 (even though matching with `Consumed` is more straightforward in a sense). So I'd say that... ye, it's not worth to apply this optimization here.